tensorflow/python/ops/nn_grad.py

```diff
@@ -809,14 +830,17 @@ def _FusedBatchNormGradGrad(op, *grad):
   """
   data_format = op.get_attr("data_format")
   epsilon = op.get_attr("epsilon")
+  is_training = op.get_attr("is_training")
   grad_y = op.inputs[0]
   x = op.inputs[1]
   scale = op.inputs[2]
+  pop_mean = op.inputs[3]
```

Comment:
I think here you will need the input 3 and 4 of op FusedBatchNorm instead of FusedBatchNormGrad. Could you forward the pop mean and pop variance to the output 3 and 4 of FusedBatchNorm in the C++ code? This way you can also unified the two branches in the "_FusedBatchNormGrad(op, *grad)".

Reply:
`pop_mean` and `pop_var` are inputs to `FusedBatchNormGrad` as well, what's the reason to not use them directly like this?

Reply:
Note that input 3 and 4 are reserve_space_1 and reserve_space_2, which are not pop mean and pop var, but 
reserve_space_1: A 1D Tensor for the computed batch mean, to be reused
                 in the gradient computation.
reserve_space_2: A 1D Tensor for the computed batch variance (inverted variance
                 in the cuDNN case), to be used in the gradient computation.

REGISTER_OP("FusedBatchNormGrad")
    .Input("y_backprop: T")
    .Input("x: T")
    .Input("scale: T")
    .Input("reserve_space_1: T")
    .Input("reserve_space_2: T")
    .Output("x_backprop: T")
    .Output("scale_backprop: T")
    .Output("offset_backprop: T")
    .Output("reserve_space_3: T")
    .Output("reserve_space_4: T")
    .Attr("T: {float}")
    .Attr("epsilon: float = 0.0001")
    .Attr("data_format: string = 'NHWC'")
    .Attr("is_training: bool = true")
......

Reply:
Ok, you did the forwarding in _FusedBatchNormGrad, instead of on the C++ side. I think that is fine too. If you go with this way, could you update the comments of reserve_space_1 and reserve_space_2 saying they are pop mean and variance when is_training is False?

Reply:
Comments were updated.